esp/esp/program/modules/handlers/classsearchmodule.py

```diff
@@ -103,8 +103,14 @@ def classsearch(self, request, tl, one, two, module, extra, prog):
             'program': self.program,
             'query': None,
         }
+        namequery = request.GET.get('namequery')
+        decoded = None
         if data is not None:
             decoded = json.loads(data)
+        elif namequery: # only search if not None and not ""
+            decoded = {'filter': 'title', 'negated': False, 'values': [namequery]}
```

Comment:
The reason is mostly that I'm lazy — the class search page proper goes through some contortions to turn what you enter into the form into a JSON query structure behind-the scenes, and pressing Enter in one of the text fields doesn't work (although I guess this probably isn't too hard to fix). And this is just a drop-dead simple form, there are fewer ways for it to break.